### PR TITLE
Add gunzip, gzip to lagom and make deflate decompression a bit faster on arm

### PR DESCRIPTION
**Pull request by nico**
This makes it easy to compare the performance of Serenity's
deflate implementation to the host system implementation.

On my M1 Max MBP:

    % time gunzip -c \
        /Users/thakis/Downloads/trace_bug.json.gz > /dev/null

takes between 0.064s and 0.082s.

    % time Build/lagom/gunzip -c \
        /Users/thakis/Downloads/trace_bug.json.gz > /dev/null

on the other hand takes 4 seconds.

---

So we're ~57x slower than macOS gunzip.

But, not to fret, after the dramatic speedups herein we only need 3.87s, merely 55x slower.

### CODE REVIEW

**Comment by nico** [comment]
(Slightly less flippantly: This reduces time in crc32 from 13% of runtime to 0.6% of runtime, so it _is_ a somewhat decent speedup of a smaller part of the computation.)

File: Userland/Libraries/LibCrypto/Checksum/CRC32.cpp [hunk 1]
```diff
@@ -11,6 +11,41 @@
 
 namespace Crypto::Checksum {
 
+#if defined(__aarch64__) && defined(__ARM_FEATURE_CRC32)
+
+void CRC32::update(ReadonlyBytes span)
+{
+    // FIXME: Does this require runtime checking on rpi?
+    //        (Maybe the instruction is present on the rpi4 but not on the rpi3?)
```
Comment (konradekk) [inline Userland/Libraries/LibCrypto/Checksum/CRC32.cpp:19]: you can check it by using feature detection from the kernel (might be necessary to propagate it to the user space).

you can refer to `Kernel/Random.cpp` to see the api; if youʼre interested in the rpi version, the drill should be the same – see `Kernel/Arch/aarch64/RPi/MMIO.cpp` instead (not really recommended).

**Review by konradekk — APPROVED** [review]
still a step forward. 👍

File: Userland/Libraries/LibCrypto/Checksum/CRC32.cpp (hunk 1 above)
Comment (nico) [reply Userland/Libraries/LibCrypto/Checksum/CRC32.cpp:19]: Do you know if the rpi3 CPU lacks this instruction while rpi4 has it? (If they both have it, then this static version is good enough.)

Comment (konradekk) [reply Userland/Libraries/LibCrypto/Checksum/CRC32.cpp:19]: v8.1 enforces the support but none of the rpi-s are apparently adhering to that standard. yet iʼm pretty sure rpi4 has it; rpi3 needs additional check.

you might try to run the qemu in rpi3 mode if you believe it emulates that faithfully—during the boot you should see feature detection results (should also be correct). i try to check scan the net…

Comment (konradekk) [reply Userland/Libraries/LibCrypto/Checksum/CRC32.cpp:19]: okay; that seems to be pretty accurate:
— https://openwrt.org/toh/raspberry_pi_foundation/raspberry_pi

boot logs suggest (heavily) that rpi3 misses it, while rpi4 has the hardware support youʼre looking for.

Comment (nico) [reply Userland/Libraries/LibCrypto/Checksum/CRC32.cpp:19]: Cool, thanks!

**Pull request merged**